core/src/main/scala/org/apache/spark/util/JsonProtocol.scala

```diff
@@ -62,18 +62,21 @@ private[spark] object JsonProtocol {
    * JSON serialization methods for SparkListenerEvents |
    * -------------------------------------------------- */
 
-  def sparkEventToJson(event: SparkListenerEvent): JValue = {
+  def sparkEventToJson(
+    event: SparkListenerEvent,
+    omitInternalAccums: Boolean = false,
+    omitUpdatedBlockStatuses: Boolean = false): JValue = {
     event match {
```

Comment:
stageSubmitted/stageCompleted/jobStart should use `omitInternalAccums`, but not jobEnd. jobEnd's interface hasn't changed. `omitUpdatedBlockStatues` is intended to be only used for taskEnd because that's when updated block statuses are reported. Thanks for catching, I will add omitInternalAccums to stageSubmitted and jobStart.